discovery_engine/test/feed_manager_test.dart

```diff
@@ -0,0 +1,179 @@
+    show Document, DocumentAdapter, DocumentFeedback;
+import 'package:xayn_discovery_engine/src/domain/models/unique_id.dart'
+    show DocumentId, StackId;
+import 'package:xayn_discovery_engine/src/domain/models/web_resource.dart'
+    show WebResource;
+import 'package:xayn_discovery_engine/src/infrastructure/box_name.dart'
+    show documentBox, activeDocumentDataBox, changedDocumentIdBox;
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_active_document_repo.dart'
+    show HiveActiveDocumentDataRepository;
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_changed_document_repo.dart'
+    show HiveChangedDocumentRepository;
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_document_repo.dart'
+    show HiveDocumentRepository;
+
+Future<void> main() async {
+  Hive.registerAdapter(DocumentAdapter());
+
+  final docBox = await Hive.openBox<Document>(documentBox, bytes: Uint8List(0));
+  final activeBox = await Hive.openBox<ActiveDocumentData>(
+    activeDocumentDataBox,
+    bytes: Uint8List(0),
+  );
+  final changedBox =
+      await Hive.openBox<Uint8List>(changedDocumentIdBox, bytes: Uint8List(0));
+
+  final engine = MockEngine();
+  const maxBatch = 5;
+  final docRepo = HiveDocumentRepository();
+  final activeRepo = HiveActiveDocumentDataRepository();
+  final changedRepo = HiveChangedDocumentRepository();
+
+  final mgr = FeedManager(engine, maxBatch, docRepo, activeRepo, changedRepo);
+
+  group('FeedManager', () {
+    final data = ActiveDocumentData(Uint8List(0));
+    final dummy = WebResource.fromJson(<String, Object>{
+      'title': 'Example',
+      'displayUrl': 'domain.com',
+      'snippet': 'snippet',
+      'url': 'http://domain.com',
+      'datePublished': '1980-01-01T00:00:00.000000',
+      'provider': <String, String>{
+        'name': 'domain',
+        'thumbnail': 'http://thumbnail.domain.com',
+      },
+    });
+    final stackId = StackId();
+    final doc2 = Document(
+      stackId: stackId,
+      personalizedRank: 2,
+      webResource: dummy,
+      isActive: true,
+    );
+    final doc3 = Document(
+      stackId: stackId,
+      personalizedRank: 3,
+      webResource: dummy,
+      isActive: false,
+    );
+    final id2 = doc2.documentId;
+    final id3 = doc3.documentId;
+    final id = DocumentId();
+
+    setUp(() async {
+      // doc2 is active & changed, doc3 is neither
+      await docRepo.updateMany([doc2, doc3]);
+      await activeRepo.update(id2, data);
+      await changedRepo.add(id2);
+
+      engine.resetCallCounter();
+    });
+
+    tearDown(() async {
+      await docBox.clear();
+      await activeBox.clear();
+      await changedBox.clear();
+
+      // reset test data
+      doc2.isActive = true;
+      doc2.feedback = DocumentFeedback.neutral;
+      doc3.isActive = false;
+      doc3.feedback = DocumentFeedback.neutral;
+      data.viewTime.clear();
```

Comment:
you could also avoid this reset by moving objects setup to `setUp` like:

```dart

group('FeedManager', () {
  late Document doc2;
  late Document doc3;

  setUp(() async {
    doc2 = Document(/*...*/);
    doc3 = Document(/*...*/);

    await docRepo.updateMany([doc2, doc3]);
    //...
  });

  tearDown(() async {
    await docBox.clear();
  });

  test('...', {
    // perform test
  });

});
```

Reply:
nice, thanks :+1: